### PR TITLE
Adding interval selection for statistics display

### PR DESCRIPTION
**Pull request by levaja**
This PR addresses issue #141 and provides option to choose time span for which statistics is displayed.

### CODE REVIEW

File: WatchDogEclipsePlugin/WatchDog/src/nl/tudelft/watchdog/logic/interval/IntervalStatistics.java [hunk 1]
```diff
@@ -21,7 +21,7 @@
 /** Gathers and calculates statistics on interval length. */
 @SuppressWarnings("javadoc")
 public class IntervalStatistics extends IntervalManagerBase {
-	private static final int FILTERED_INTERVALS_IN_MINUTES = 60;
+	private static final int FILTERED_INTERVALS_IN_MINUTES = 600;
```
Comment (Inventitech) [inline WatchDogEclipsePlugin/WatchDog/src/nl/tudelft/watchdog/logic/interval/IntervalStatistics.java]: Please add an explanation for how this number.

File: WatchDogEclipsePlugin/WatchDog/src/nl/tudelft/watchdog/logic/interval/IntervalStatistics.java [hunk 2]
```diff
@@ -41,8 +41,23 @@
...
+		}
+	}
+
+	private StatisticsInterval selectedInterval;
+
 	/** Constructor. */
-	public IntervalStatistics(IntervalManager intervalManager) {
+	public IntervalStatistics(IntervalManager intervalManager,
+			StatisticsInterval selectedInterval) {
+		this.selectedInterval = selectedInterval;
```
Comment (Inventitech) [inline WatchDogEclipsePlugin/WatchDog/src/nl/tudelft/watchdog/logic/interval/IntervalStatistics.java]: Please retain assignment order.

File: WatchDogEclipsePlugin/WatchDog/src/nl/tudelft/watchdog/logic/interval/IntervalStatistics.java [hunk 3]
```diff
@@ -57,8 +72,42 @@ private void addIntervals(IntervalManager intervalManager) {
 		intervals.addAll(intervalManager.getOpenIntervals());
 	}
 
-	/** Filters out and removes intervals which are older than one hour. */
+	/**
+	 * Filters out and removes intervals which are older than 10 hours from
+	 * Database. Filters intervals for selected time span.
+	 */
 	private void filterIntervals() {
+		int numberOfMinutes = 600;
```
Comment (Inventitech) [inline WatchDogEclipsePlugin/WatchDog/src/nl/tudelft/watchdog/logic/interval/IntervalStatistics.java]: This is a magic number (why not use FILTERED_INTERVALS_IN_MINUTES?)

File: WatchDogEclipsePlugin/WatchDog/src/nl/tudelft/watchdog/logic/interval/IntervalStatistics.java [hunk 4]
```diff
@@ -57,8 +72,42 @@ private void addIntervals(IntervalManager intervalManager) {
...
+			numberOfMinutes = 300;
+			break;
+
+		case HOURS_8:
+			numberOfMinutes = 480;
+			break;
+
+		case HOURS_10:
+			numberOfMinutes = 600;
+		}
```
Comment (Inventitech) [inline WatchDogEclipsePlugin/WatchDog/src/nl/tudelft/watchdog/logic/interval/IntervalStatistics.java]: By integrating numberOfMinutes into StatisticsInterval, this is redundant.

File: WatchDogEclipsePlugin/WatchDog/src/nl/tudelft/watchdog/logic/interval/IntervalStatistics.java [hunk 5]
```diff
@@ -67,21 +116,30 @@ private void filterIntervals() {
...
+
 		for (IntervalBase interval : intervals) {
-			if (interval.getEnd().after(thresholdDate.toDate())) {
-				IntervalBase clonedInterval = (IntervalBase) interval.clone();
-				if (interval.getStart().before(thresholdDate.toDate())) {
-					clonedInterval.setStartTime(thresholdDate.toDate());
-				}
-				if (!clonedInterval.isClosed()) {
-					clonedInterval.setEndTime(mostRecentDate);
+			// Filtering intervals from the last 10 hours
```
Comment (Inventitech) [inline WatchDogEclipsePlugin/WatchDog/src/nl/tudelft/watchdog/logic/interval/IntervalStatistics.java]: please remove phrase 10 hours (can vary by modifying a constant in a different place and it's not obvious to update the comment here)

File: WatchDogEclipsePlugin/WatchDog/src/nl/tudelft/watchdog/logic/interval/IntervalStatistics.java (hunk 1 above)
Comment (levaja) [inline WatchDogEclipsePlugin/WatchDog/src/nl/tudelft/watchdog/logic/interval/IntervalStatistics.java]: Fixed.

File: WatchDogEclipsePlugin/WatchDog/src/nl/tudelft/watchdog/logic/interval/IntervalStatistics.java (hunk 2 above)
Comment (levaja) [inline WatchDogEclipsePlugin/WatchDog/src/nl/tudelft/watchdog/logic/interval/IntervalStatistics.java]: Fixed.

File: WatchDogEclipsePlugin/WatchDog/src/nl/tudelft/watchdog/logic/interval/IntervalStatistics.java (hunk 3 above)
Comment (levaja) [inline WatchDogEclipsePlugin/WatchDog/src/nl/tudelft/watchdog/logic/interval/IntervalStatistics.java]: Fixed, good point!

File: WatchDogEclipsePlugin/WatchDog/src/nl/tudelft/watchdog/logic/interval/IntervalStatistics.java [hunk 6]
```diff
@@ -67,21 +116,30 @@ private void filterIntervals() {
...
+			if (interval.getEnd().after(thresholdDateDatabase.toDate())) {
```
Comment (Inventitech) [inline WatchDogEclipsePlugin/WatchDog/src/nl/tudelft/watchdog/logic/interval/IntervalStatistics.java]: Please simplify this piece of code by reducing the nesting depth, creating functions with tell-tale names and also creating a unit test for it.

It is quite difficult to see that the only modification you made is introducing if (interval.getEnd().after(thresholdDateView.toDate())).

File: WatchDogEclipsePlugin/WatchDog/src/nl/tudelft/watchdog/ui/WatchDogView.java [hunk 7]
```diff
@@ -176,6 +184,27 @@ public void widgetDefaultSelected(SelectionEvent e) {
...
+				Combo widget = (Combo) e.getSource();
+				selectedInterval = StatisticsInterval.values()[widget
+						.getSelectionIndex()];
+			}
+
+			@Override
+			public void widgetDefaultSelected(SelectionEvent e) {
+			}
+
+		}, new String[] { "10 minutes.", "30 minutes.", "1 hour.", "2 hours.",
```
Comment (Inventitech) [inline WatchDogEclipsePlugin/WatchDog/src/nl/tudelft/watchdog/ui/WatchDogView.java]: Why not supply these values to the enum IntervalStatistics and then call IntervalStatistics.values()?

File: WatchDogEclipsePlugin/WatchDog/src/nl/tudelft/watchdog/logic/interval/IntervalStatistics.java (hunk 4 above)
Comment (levaja) [inline WatchDogEclipsePlugin/WatchDog/src/nl/tudelft/watchdog/logic/interval/IntervalStatistics.java]: Great suggestion, thanks! Just another field in enum StatisticsInterval makes everything more elegant.

File: WatchDogEclipsePlugin/WatchDog/src/nl/tudelft/watchdog/logic/interval/IntervalStatistics.java [hunk 8]
```diff
@@ -67,21 +116,30 @@ private void filterIntervals() {
...
+						clonedInterval.setStartTime(thresholdDateView.toDate());
+					}
+					if (!clonedInterval.isClosed()) {
+						clonedInterval.setEndTime(mostRecentDate);
+					}
+					filteredIntervals.add(clonedInterval);
 				}
-				filteredIntervals.add(clonedInterval);
 			} else {
+				// Remove from Database intervals older than 10 hours
```
Comment (Inventitech) [inline WatchDogEclipsePlugin/WatchDog/src/nl/tudelft/watchdog/logic/interval/IntervalStatistics.java]: please remove phrase 10 hours (can vary by modifying a constant in a different place and it's not obvious to update the comment here)

File: WatchDogEclipsePlugin/WatchDog/src/nl/tudelft/watchdog/logic/interval/IntervalStatistics.java (hunk 5 above)
Comment (levaja) [inline WatchDogEclipsePlugin/WatchDog/src/nl/tudelft/watchdog/logic/interval/IntervalStatistics.java]: Fixed.

File: WatchDogEclipsePlugin/WatchDog/src/nl/tudelft/watchdog/ui/util/UIUtils.java [hunk 9]
```diff
@@ -274,4 +275,14 @@ public static Link createLinkedLabel(Composite parent,
 		return link;
 	}
 
+	/** Creates a Combo List of String items. */
```
Comment (Inventitech) [inline WatchDogEclipsePlugin/WatchDog/src/nl/tudelft/watchdog/ui/util/UIUtils.java]: I think there is one other Combo box used in WatchDog in UserRegistrationPage.java. Could you make that use your new UIUtils function? :)

File: WatchDogEclipsePlugin/WatchDog/src/nl/tudelft/watchdog/logic/interval/IntervalStatistics.java (hunk 8 above)
Comment (levaja) [inline WatchDogEclipsePlugin/WatchDog/src/nl/tudelft/watchdog/logic/interval/IntervalStatistics.java]: Fixed.

File: WatchDogEclipsePlugin/WatchDog/src/nl/tudelft/watchdog/ui/WatchDogView.java (hunk 7 above)
Comment (levaja) [inline WatchDogEclipsePlugin/WatchDog/src/nl/tudelft/watchdog/ui/WatchDogView.java]: Good point, fixed.

File: WatchDogEclipsePlugin/WatchDog/src/nl/tudelft/watchdog/ui/util/UIUtils.java (hunk 9 above)
Comment (levaja) [inline WatchDogEclipsePlugin/WatchDog/src/nl/tudelft/watchdog/ui/util/UIUtils.java]: Done!

**Comment by Inventitech** [comment]
Thanks for providing this PR. I fixed a few minor issues ("Showing ... intervals from ... to ...", even when no intervals were shown, visibility of member fields, field id was redundant, some comments were trivial) and refactored the loop a bit to make it more readable.

**Comment by levaja** [comment]
Thanks for fixing this, I've checked that all. It's good feedback for me to learn for future tasks :+1:

**Pull request merged**